src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -1762,6 +1762,8 @@ private void getDestinationMetadata() throws SQLServerException {
                     } finally {
                         DESTINATION_COL_METADATA_LOCK.unlock();
                     }
+                } else {
+                    DESTINATION_COL_METADATA_LOCK.unlock();
```

Comment:
A better pattern would be to move the start of the try block to the line below the lock(). That way you have one lock and one unlock and you are sure to always unlock in the finally after taking the lock.